torch_xla/experimental/checkpoint.py

```diff
@@ -0,0 +1,205 @@
+import io
+import torch
+import torch_xla
+import torch_xla.core.xla_model as xm
+import torch_xla.experimental.xla_sharding as xs
+
+from torch.distributed.checkpoint._traverse import set_element
+from torch.distributed.checkpoint.default_planner import (
+    create_default_local_load_plan,
+    create_default_global_load_plan,
+)
+from torch.distributed.checkpoint.planner import (
+    SavePlanner,
+    LoadPlanner,
+    SavePlan,
+    LoadPlan,
+    ReadItem,
+    WriteItem,
+    WriteItemType,
+    LoadItemType,
+)
+from torch.distributed.checkpoint.planner_helpers import (
+    create_read_items_for_chunk_list,)
+from torch.distributed.checkpoint.metadata import (
+    BytesStorageMetadata,
+    ChunkStorageMetadata,
+    TensorStorageMetadata,
+    MetadataIndex,
+    Metadata,
+    STATE_DICT_TYPE,
+    STORAGE_TYPES,
+)
+from torch.distributed.checkpoint._nested_dict import (
+    FLATTEN_MAPPING,
+    flatten_state_dict,
+)
+from torch.distributed.checkpoint.utils import find_state_dict_object
+from torch.utils._pytree import tree_map
+from torch_xla.experimental.xla_sharding import (XLAShardedTensor, XLAShard,
+                                                 ShardingType)
+from typing import List, Tuple, Dict
+
+__all__ = [
+    "XLASavePlanner",
+    "XLALoadPlanner",
+]
+
+
+class XLASavePlanner(SavePlanner):
+  """
+  XLASavePlanner provides an implementation of the SavePlanner interface
+  which handles state_dicts containing XLAShardedTensor.
+  """
+
+  def set_up_planner(self, state_dict: STATE_DICT_TYPE,
+                     is_coordinator: bool) -> None:
+    raise NotImplemented
+
+  def create_local_plan(self) -> SavePlan:
+    raise NotImplemented
+
+  def create_global_plan(
+      self, all_plans: List[SavePlan]) -> Tuple[List[SavePlan], Metadata]:
+    raise NotImplemented
+
+  def finish_plan(self, new_plan: SavePlan) -> SavePlan:
+    raise NotImplemented
+
+
+class XLALoadPlanner(LoadPlanner):
```

Comment:
Good point! I'll move the declarations into `__init__`. `set_up_planner` will still handle initializing most of the fields, but this will definitely help with readability.